### PR TITLE
Added Curse of Madness to Summon Event pool

### PR DESCRIPTION
**Pull request by coiax**
:cl: coiax
add: Curse of Madness can now be triggered by a wizard's Summon Events,
at the same chance as Summon Guns or Summon Magic.
admin: When an admin triggers Curse of Madness manually, they can
specify their own dark truth to horrify the station with.
/:cl:

Random Curse of Madness events pick a line from the redpill.json file,
which I've added some more entries to.

### CODE REVIEW

**Comment by Therapist1** [comment]
“What happened aboard Kosmicheskaya Stantsiya?”

**Comment by coiax** [comment]
Everyone knows what happened, it's why they don't talk about it.

File: code/modules/events/wizard/madness.dm [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+	typepath = /datum/round_event/wizard/madness
+	earliest_start = 0 MINUTES
+
+	var/forced_secret
+
+/datum/round_event_control/wizard/madness/admin_setup()
+	if(!check_rights(R_FUN))
+		return
+
+	var/suggested = pick(strings("redpill.json", "redpill_questions"))
```
Comment (Cyberboss) [inline code/modules/events/wizard/madness.dm]: String duplication, these should be defined somewhere

File: code/modules/events/wizard/madness.dm [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+	forced_secret = input(usr, "What horrifying truth will you reveal?", "Curse of Madness", suggested) as text
```
Comment (Cyberboss) [inline code/modules/events/wizard/madness.dm]:
```suggestion
 	forced_secret = (input(usr, "What horrifying truth will you reveal?", "Curse of Madness", suggested) as text) || suggested
```

File: code/modules/events/wizard/madness.dm [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+/datum/round_event/wizard/madness/start()
+	var/datum/round_event_control/wizard/madness/C = control
+
+	var/horrifying_truth
+
+	if(C.forced_secret)
+		horrifying_truth = C.forced_secret
+		C.forced_secret = null
+	else
+		horrifying_truth = pick(strings("redpill.json", "redpill_questions"))
```
Comment (Cyberboss) [inline code/modules/events/wizard/madness.dm]: And then just line 23

File: code/modules/events/wizard/madness.dm (hunk 1 above)
Comment (Cyberboss) [reply code/modules/events/wizard/madness.dm]: Also, given that the second parameter is `directory` how does this even work? Does the original?

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/modules/events/wizard/madness.dm (hunk 3 above)
Comment (coiax) [reply code/modules/events/wizard/madness.dm]: But sometimes the event will just trigger without an admin pushing the button, so we'd still need to handle if forced_secret was null

File: code/modules/events/wizard/madness.dm (hunk 1 above)
Comment (optimumtact) [reply code/modules/events/wizard/madness.dm]: read the code

**Pull request merged**